src/main/java/com/tabnine/binary/fetch/BinaryVersionFetcher.java

```diff
@@ -27,6 +28,12 @@ public BinaryVersionFetcher(LocalBinaryVersions localBinaryVersions, BinaryRemot
      * @throws SecurityException, NoExistingBinaryException if something went wrong
      */
     public String fetchBinary() throws NoValidBinaryToRunException {
+        Optional<BinaryVersion> bootstrappedVersion = BootstrapperSupport.bootstrap_version(localBinaryVersions, binaryRemoteSource, binaryDownloader);
```

Comment:
I think we can leverage `orElseGet` here after the refactor

Reply:
the orelse part is all the existing code